core/app/[locale]/(default)/(faceted)/brand/[slug]/page.tsx

```diff
@@ -80,229 +47,185 @@ const createBrandSearchParamsCache = cache(async (props: Props) => {
   const { slug } = await props.params;
 
-  const variables = cachedBrandDataVariables(slug);
-  const data = await getBrandPageData(variables);
-
-  return data.settings?.storefront.catalog?.productComparisonsEnabled ?? false;
-}
+  const brandId = Number(slug);
 
-const cachedCompareProductIds = cache(async (props: Props) => {
-  const searchParams = await props.searchParams;
+  const { brand } = await getBrandPageData(brandId);
 
-  const compareLoader = createCompareLoader();
-
-  const { compare } = compareLoader(searchParams);
-
-  return { entityIds: compare ? compare.map((id: string) => Number(id)) : [] };
-});
-
-async function getCompareProducts(props: Props) {
-  const compareIds = await cachedCompareProductIds(props);
-
-  const products = await getCompareProductsData(compareIds);
-
-  return products.map((product) => ({
-    id: product.entityId.toString(),
-    title: product.name,
-    image: product.defaultImage
-      ? { src: product.defaultImage.url, alt: product.defaultImage.altText }
-      : undefined,
-    href: product.path,
-  }));
-}
+  if (!brand) {
+    return notFound();
+  }
 
-async function getFilterLabel(): Promise<string> {
-  const t = await getTranslations('Faceted.FacetedSearch');
+  const { pageTitle, metaDescription, metaKeywords } = brand.seo;
 
-  return t('filters');
+  return {
+    title: pageTitle || brand.name,
+    description: metaDescription,
+    keywords: metaKeywords ? metaKeywords.split(',') : null,
+  };
 }
 
-async function getSortLabel(): Promise<string> {
-  const t = await getTranslations('Faceted.SortBy');
+export default async function Brand(props: Props) {
+  const { locale, slug } = await props.params;
 
-  return t('sortBy');
-}
+  setRequestLocale(locale);
 
-async function getCompareLabel(): Promise<string> {
-  const t = await getTranslations('Components.ProductCard.Compare');
+  const t = await getTranslations('Faceted');
 
-  return t('compare');
-}
+  const brandId = Number(slug);
 
-async function getRemoveLabel(): Promise<string> {
-  const t = await getTranslations('Components.ProductCard.Compare');
+  const { brand, settings } = await getBrandPageData(brandId);
 
-  return t('remove');
-}
+  if (!brand) {
+    return notFound();
+  }
 
-async function getMaxCompareLimitMessage(): Promise<string> {
-  const t = await getTranslations('Components.ProductCard.Compare');
+  const productComparisonsEnabled =
+    settings?.storefront.catalog?.productComparisonsEnabled ?? false;
 
-  return t('maxCompareLimit');
-}
+  const streamableFacetedSearch = Streamable.from(async () => {
+    const searchParams = await props.searchParams;
+    const customerAccessToken = await getSessionCustomerAccessToken();
+    const currencyCode = await getPreferredCurrencyCode();
 
-async function getEmptyStateTitle(): Promise<string> {
-  const t = await getTranslations('Faceted.Brand.Empty');
+    const searchParamsCache = await createBrandSearchParamsCache(slug);
```

Comment:
Creating a search params cache like this seems worthless since we can't access this cache and thus read set values on other RSCs. If we're instead looking to parse the search params,`createLoeader` seems more appropriate